src/NServiceBus.NHibernate/Internal/ConfigureNHibernate.cs

```diff
@@ -20,57 +19,62 @@ namespace NServiceBus.Persistence.NHibernate
-
-        private const string Message =
-            @"
-To run NServiceBus with NHibernate you need to at least specify the database connectionstring.
-Here is an example of what is required:
-  <appSettings>
-    <!-- dialect is defaulted to MsSql2008Dialect, if needed change accordingly -->
-    <add key=""NServiceBus/Persistence/NHibernate/dialect"" value=""NHibernate.Dialect.{your dialect}""/>
-
-    <!-- other optional settings examples -->
-    <add key=""NServiceBus/Persistence/NHibernate/connection.provider"" value=""NHibernate.Connection.DriverConnectionProvider""/>
-    <add key=""NServiceBus/Persistence/NHibernate/connection.driver_class"" value=""NHibernate.Driver.Sql2008ClientDriver""/>
-    <!-- For more setting see http://www.nhforge.org/doc/nh/en/#configuration-hibernatejdbc and http://www.nhforge.org/doc/nh/en/#configuration-optional -->
-  </appSettings>
-  
-  <connectionStrings>
-    <!-- Default connection string for all NHibernate/Sql persisters -->
-    <add name=""NServiceBus/Persistence"" connectionString=""Data Source=.\SQLEXPRESS;Initial Catalog=nservicebus;Integrated Security=True"" />
-    
-    <!-- Optional overrides per persister -->
-    <add name=""NServiceBus/Persistence/NHibernate/Timeout"" connectionString=""Data Source=.\SQLEXPRESS;Initial Catalog=timeout;Integrated Security=True"" />
-    <add name=""NServiceBus/Persistence/NHibernate/Saga"" connectionString=""Data Source=.\SQLEXPRESS;Initial Catalog=sagas;Integrated Security=True"" />
-    <add name=""NServiceBus/Persistence/NHibernate/Subscription"" connectionString=""Data Source=.\SQLEXPRESS;Initial Catalog=subscription;Integrated Security=True"" />
-    <add name=""NServiceBus/Persistence/NHibernate/Deduplication"" connectionString=""Data Source=.\SQLEXPRESS;Initial Catalog=gateway;Integrated Security=True"" />
-    <add name=""NServiceBus/Persistence/NHibernate/Distributor"" connectionString=""Data Source=.\SQLEXPRESS;Initial Catalog=distributor;Integrated Security=True"" />
-  </connectionStrings>";
-
-        static readonly ILog Logger = LogManager.GetLogger(typeof(ConfigureNHibernate));
         static readonly Regex PropertyRetrievalRegex = new Regex(@"NServiceBus/Persistence/NHibernate/([\W\w]+)$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
         private static ConnectionStringSettingsCollection connectionStringSettingsCollection;
-
         public const string DefaultDialect = "NHibernate.Dialect.MsSql2008Dialect";
 
+        readonly Configuration configuration;
+
+        public ConfigureNHibernate(ReadOnlySettings settings, string connectionStringKeySuffix, params string[] settingsKeys)
+        {
+            if (connectionStringKeySuffix == null)
+            {
+                throw new ArgumentNullException("connectionStringKeySuffix");
+            }
+            if (settingsKeys.Length == 0)
+            {
+                throw new InvalidOperationException("At least one setting key is required");
```

Comment:
The new exceptions are raised by `ValidateConfigurationViaConfigFile` and `ValidateConfigurationViaCode` methods to give the user some idea of what to do. This is still not ideal because if a user missed adding `NServiceBus/Persistence` connection string, he would be given three most popular config options in order of precedence but 95% of the time he should go with option 2